### DISCUSSION TITLE
Not sure if it's a false detection: Ansible doc has an example of a certificate

### DISCUSSION THREAD

**Discussion by synergiator** [aquasecurity/trivy#7504] [Q&A]
### IDs

HIGH - Asymmetric private Key

### Description

Maybe it's a false positive - maybe not?
Maybe related to https://github.com/aquasecurity/trivy/discussions/6097?
I have posted the issue to the Ansible community.
https://github.com/ansible-collections/community.general/issues/8862

### Reproduction Steps

```bash
Install Ansible Galaxy general collection and run Trivy on it.

===========================================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)
HIGH: AsymmetricPrivateKey (private-key)
════════════════════════════════════════
Asymmetric Private Key
────────────────────────────────────────
 /root/.ansible/collections/ansible_collections/community/general/plugins/modules/java_keystore.py:157 (added by 'RUN set -ex && chmod +x ./*.sh &&     ls')
────────────────────────────────────────
 155         -----END CERTIFICATE-----
 156       private_key: |
 157 [ ----BEGIN RSA PRIVATE KEY-----************************************************************************************************************************************************-----END RSA PRIVATE
 158       password: changeit
```

### Target

Container Image

### Scanner

Secret

### Target OS

Alpine

### Debug Output

```bash
no time to test
```

### Version

```bash
Version: 0.55.1
```

### Checklist

- [ ] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [ ] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by felixfontein** [comment]
It's a false positive in the sense that the key is only part of the documentation and not used by anything in the Ansible collection.

**Comment by DmitriyLewen** [comment]
Hello @synergiator @felixfontein 
Thanks for your report!

Trivy uses [builtinAllowRules](https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/secret/builtin-allow-rules.go) to avoid false positives for test, examples, etc.
unfortunately, in your case there is no option to understand that this is a test Asymmetric private Key.

Regards, Dmitriy

**Comment by PT-GD** [comment]
@DmitriyLewen an easy fix would be to update https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/secret/builtin-rules.go#L188 to require a potentially matching secret to be a "sane" minimal length for an RSA private key. Anything less than 64 characters is too little base64 to even represent a 384-bit RSA private key (the smallest any known library implementations even notionally support). You would typically expect 340 characters for a 2048-bit private key.

Right now you match on any length string (including zero length) between the BEGIN and END block.

**Comment by PT-GD** [comment]
@synergiator: Feel free to steal my verbiage from https://github.com/aquasecurity/trivy/discussions/8601 in updating the issue text to reflect that this is confirmed as a false positive. The rule still matches even after the Ansible project's change to redact the example key body in their documentation.

I closed my issue as a duplicate of yours.

**Comment by PT-GD** [comment]
I opened #8602 to attempt to address this, as it looks like @DmitriyLewen already had an open, unclaimed issue #7700 on the topic.
